Userland/Libraries/LibCore/Group.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2022, Kenneth Myhra <kennethmyhra@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/Group.h>
+#include <LibCore/System.h>
+#include <ctype.h>
+
+namespace Core {
+
+ErrorOr<Optional<Group>> Group::from_name(StringView group_name)
+{
+    auto group = TRY(Core::System::getgrnam(group_name));
+    if (!group.has_value()) {
+        return Optional<Group> {};
+    }
+
+    return TRY(from_libc_group(group.value()));
+}
+
+ErrorOr<Optional<Group>> Group::from_gid(gid_t group_id)
+{
+    auto group = TRY(Core::System::getgrgid(group_id));
+    if (!group.has_value()) {
+        return Optional<Group> {};
+    }
+
+    return TRY(from_libc_group(group.value()));
+}
+
+#ifndef AK_OS_MACOS
+ErrorOr<void> Group::add_group(Group& group)
+{
+    if (group.group_name().is_null_or_empty())
+        return Error::from_string_literal("Group::group_name() is required.");
```

Comment:
That's kind of a weird error message. Wouldn't `"Group name must not be empty."` / `"Group name is empty."` be better?